Mathlib/LinearAlgebra/SpecialLinearGroup.lean

```diff
@@ -0,0 +1,342 @@
+/-
+Copyright (c) 2025 Antoine Chambert-Loir. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir
+-/
+
+import Mathlib.LinearAlgebra.Determinant
+import Mathlib.LinearAlgebra.Dual.Basis
+import Mathlib.LinearAlgebra.Matrix.Dual
+import Mathlib.LinearAlgebra.GeneralLinearGroup
+import Mathlib.LinearAlgebra.Charpoly.BaseChange
+
+/-!
+# The special linear group of a module
+
+If `R` is a commutative ring and `V` is an `R`-module,
+we define `SpecialLinearGroup R V` as the subtype of
+linear equivalences `V ≃ₗ[R] V` with determinant 1.
+When `V` doesn't have a finite basis, the determinant is defined by 1
+and the definition gives `V ≃ₗ[R] V`.
+The interest of this definition is that `SpecialLinearGroup R V`
+has a group structure. (Starting from linear maps wouldn't have worked.)
+
+The file is constructed parallel to the one defining `Matrix.SpecialLinearGroup`.
+
+We provide `SpecialLinearGroup.toLinearEquiv`: the canonical map
+from `SpecialLinearGroup R V` to `V ≃ₗ[R] V`, as a monoid hom.
+
+When `V` is free and finite over `R`, we define
+* `SpecialLinearGroup.dualMap`
+* `SpecialLinearGroup.baseChange`
+
+We define `Matrix.SpecialLinaerGruop.toLin'_equiv`: the mul equivalence
+from `Matrix.SpecialLinearGroup n R` to `SpecialLinearGroup R (n → R)`
+and its variant
+`Matrix.SpecialLinearGroup.toLin_equiv`,
+from `Matrix.SpecialLinearGroup n R` to `SpecialLinearGroup R V`,
+associated with a finite basis of `V`.
+
+-/
+
+variable {R V : Type*} [CommRing R] [AddCommGroup V] [Module R V]
+
+variable (R V) in
+/-- The special linear group of a module.
+
+This is only meaningful when the module is finite and free,
+for otherwise, it coincides with the group of linear equivalences. -/
+abbrev SpecialLinearGroup := { u : V ≃ₗ[R] V // u.det = 1 }
+
+namespace SpecialLinearGroup
+
+theorem det_eq_one (u : SpecialLinearGroup R V) :
+    LinearMap.det (u : V →ₗ[R] V) = 1 := by
+  simp [← LinearEquiv.coe_det, u.prop]
+
+instance : CoeFun (SpecialLinearGroup R V) (fun _ ↦ V → V) where
+  coe u x := u.val x
+
+theorem ext_iff (u v : SpecialLinearGroup R V) : u = v ↔ ∀ x : V, u x = v x := by
+  simp only [← Subtype.coe_inj, LinearEquiv.ext_iff]
+
+@[ext]
+theorem ext (u v : SpecialLinearGroup R V) : (∀ x, u x = v x) → u = v :=
+  (SpecialLinearGroup.ext_iff u v).mpr
+
+section rankOne
+
+variable [Nontrivial R] [Module.Free R V] [Module.Finite R V]
+
+-- Move to other file ?
```

Comment:
Yes please, move it to one file in `Mathlib.LinearAlgebra.FreeModule.Finite`. Currently there is no single file that works, but adding an import there is not a big deal.